src/gt4py/next/iterator/transforms/global_tmps.py

```diff
@@ -103,20 +169,58 @@ def _transform_by_pattern(
             ) = type_info.apply_to_primitive_constituents(
                 type_info.extract_dtype,
                 tmp_expr.type,
-                tuple_constructor=lambda *elements: tuple(elements),
+                tuple_constructor=lambda *elements: elements,
             )
 
-            # allocate temporary for all tuple elements
-            def allocate_temporary(tmp_name: str, dtype: ts.ScalarType):
-                declarations.append(itir.Temporary(id=tmp_name, domain=domain_expr, dtype=dtype))  # noqa: B023 # function only used inside loop
+            tmp_domains: SymbolicDomain | tuple[SymbolicDomain | tuple, ...] = tmp_expr.annex.domain
+
+            if cpm.is_applied_as_fieldop(tmp_expr):
+                # In this case all tuple elements have the same size (or will be `NEVER`).
+                # Create the tuple structure with that domain.
+                domains = list(
+                    set(next_utils.flatten_nested_tuple((tmp_domains,)))
+                    - set([infer_domain.DomainAccessDescriptor.NEVER])  # type: ignore[list-item] # type should always be `SymbolicDomain`
```

Comment:
There is a simpler way to create sets with specific elements:
```suggestion
                    - {infer_domain.DomainAccessDescriptor.NEVER}  # type: ignore[list-item] # type should always be `SymbolicDomain`
```

Reply:
ups